src/test/java/org/marketdesignresearch/mechlib/mechanism/auctions/cca/CCATest.java

```diff
@@ -258,11 +264,12 @@ public void testStepByStepCCA() {
         // Second round
         for (Bidder bidder : domain.getBidders()) {
             Bundle bestBundle = bidder.getBestBundle(cca.getCurrentPrices());
-            BundleBid bundleBid = new BundleBid(cca.getCurrentPrices().getPrice(bestBundle).getAmount(), bestBundle, UUID.randomUUID().toString());
-            Bid bid = new Bid(Sets.newHashSet(bundleBid));
-            checkBidEquality(bid, cca.proposeBid(bidder));
+            BundleValuePair bundleBid = new BundleValuePair(cca.getCurrentPrices().getPrice(bestBundle).getAmount(), bestBundle, UUID.randomUUID().toString());
+            BundleValueBid<BundleValuePair> bid = new BundleValueBid<>(Sets.newHashSet(bundleBid));
+            DemandQuery query = (DemandQuery) cca.getCurrentInteraction(bidder);
+            assertEquals(bestBundle, query.proposeDemandBid().getDemandedBundle());
             // Submit bids one by one
-            cca.submitBid(bidder, bid);
+            query.submitDemandBid(new DemandBid(bestBundle));
```

Comment:
Example on how to use the new interaction interface

Reply:
I didn't go through it in detail, but it seems like you've already handled the open question of generics & type-casts - looks very clean this way, and I'm looking forward to seeing the final first version of this. Happy to discuss it or review it in detail when you think it's a good time for this.

Reply:
Yes, I'm quite happy with the current solution.